src/py42/clients/orgs.py

```diff
@@ -1,8 +1,225 @@
+
+
+class OrgSettingsManager(object):
+    def __init__(self, org_client, org_id):
+        org_response = org_client.get_by_id(org_id, incSettings=True)
+        org_settings_response = org_client.get_settings_by_id(org_id)
+
+        org_dict = org_response.data
+        settings_dict = org_dict.pop("settings")
+
+        self._org = ChainMap({}, org_dict)
+        self._settings = ChainMap({}, settings_dict)
+        self._t_settings = ChainMap({}, org_settings_response.data)
+        self._org_client = org_client
+        self.errored = None
+        self.org_response = None
+        self.org_settings_response = None
+
+    @property
+    def changes(self):
+        changes_dict = {}
+        changes_dict["settings"] = self._diff_chainmap(self._org) or {}
+        changes_dict["settings"].update(self._diff_chainmap(self._settings))
+        t_settings_diff = list(self._diff_chainmap(self._t_settings).values())
+        changes_dict["t_settings"] = t_settings_diff or None
+        return changes_dict
+
+    @property
+    def org_id(self):
+        return self._org["orgId"]
+
+    @property
+    def org_name(self):
+        return self._org["orgName"]
+
+    @org_name.setter
+    def org_name(self, name):
+        self._org["orgName"] = name
+
+    @property
+    def external_reference(self):
+        return self._org["orgExtRef"]
+
+    @external_reference.setter
+    def external_reference(self, value):
+        self._org["orgExtRef"] = value
+
+    @property
+    def notes(self):
+        return self._org["notes"]
+
+    @notes.setter
+    def notes(self, value):
+        self._org["notes"] = value
+
+    @property
+    def archive_hold_days(self):
+        return self._settings["archiveHoldDays"]
+
+    @archive_hold_days.setter
+    def archive_hold_days(self, value):
+        self._settings["archiveHoldDays"] = value
+        self._settings["isUsingQuotaDefaults"] = False
+
+    @property
+    def max_user_subscriptions(self):
+        return self._settings["maxSeats"]
+
+    @max_user_subscriptions.setter
+    def max_user_subscriptions(self, value):
+        self._settings["maxSeats"] = value
+        self._settings["isUsingQuotaDefaults"] = False
+
+    @property
+    def endpoint_monitoring_enabled(self):
+        value = self._t_settings["org-securityTools-enable"]["value"]
+        return str_to_bool(value)
+
+    @endpoint_monitoring_enabled.setter
+    @bool_required
+    def endpoint_monitoring_enabled(self, value):
+        self._t_settings["org-securityTools-enable"] = {
+            "key": "org-securityTools-enable",
+            "value": bool_to_str(value),
+            "locked": False,
+        }
+        self._t_settings["device_advancedExfiltrationDetection_enabled"] = {
+            "key": "device_advancedExfiltrationDetection_enabled",
+            "value": bool_to_str(value),
+            "locked": False,
+        }
+        if not value:  # disable everything but FMC, like the UI does
```

Comment:
Would it make sense to remove this `if` condition and just have the `self.endpoint_monitoring_removable_media_enabled = value` etc ?

Reply:
These are a bit of an oddity, since it's trying to somewhat mimic the UI checkboxes for these settings. In the console UI, if you click the "enable endpoint monitoring" box, it auto-checks all the other boxes, same if you uncheck it. 

Since those flags are tied together in the UI, there's no way to have "endpoint monitoring" disabled, but also have "removable media" enabled. But you can craft an API call where they are that way. So I have it where if you set the `removeable_media_enabled` attribute to `True`, it sets `endpoint_monitoring_enabled` to true (since it has to be). But if `endpoint_monitoring_enabled = True` means _everything_ gets set to true then we'd have to have another method of setting _just_ the values (there are two) which coincide with the endpoint monitoring checkbox itself in the UI. 

So it's a slight departure from the UI, in that `endpoint_monitoring_enabled = True` only sets the minimum set of keys, where `endpoint_monitoring_enabled = False` sets everything to `False`, but I think it is a good balance between prevent incorrect states while also making the user explicitly set the things they want to be on.